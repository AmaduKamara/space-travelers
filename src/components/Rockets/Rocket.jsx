import React from 'react';
import PropTypes from 'prop-types';

function Rocket({
  id,
  rocket_name: name,
  rocket_image: rocketImage,
  description,
  reserved,
  handleToggleClick,
}) {
  return (
    <div className="flex mb-6">
      <img src={rocketImage} alt="rocket" width="250" />
      <div className="mx-5">
        <h2 className="text-xl font-semibold pt-2">{name}</h2>
        <p className="my-4">
          {reserved && (
            <span className="bg-green-600 text-white py-1 px-3 rounded-full">
              Reserved
            </span>
          )}
          {' '}
          {description}
        </p>
        {reserved ? (
          <button
            type="button"
            className="py-2 px-5 rounded text-red-500 border border-red-400"
            onClick={() => handleToggleClick(id)}
          >
            Cancel Reserve
          </button>
        ) : (
          <button
            type="button"
            className="py-2 px-5 rounded text-white bg-blue-500 "
            onClick={() => handleToggleClick(id)}
          >
            Reserve Rocket
          </button>
        )}
      </div>
    </div>
  );
}

Rocket.propTypes = {
  id: PropTypes.number.isRequired,
  rocket_name: PropTypes.string.isRequired,
  rocket_image: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  handleToggleClick: PropTypes.func.isRequired,
  reserved: PropTypes.bool,
};

Rocket.defaultProps = {
  reserved: false,
};
export default Rocket;
