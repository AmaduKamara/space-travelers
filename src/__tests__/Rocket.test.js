import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import Rocket from '../components/Rockets/Rocket';

const renderer = require('react-test-renderer');

it('should do render reserve rocket button and match the snapshot', () => {
  const id = 1;
  const rocketName = 'Test Rocket';
  const rocketImage = 'img';
  const description = 'Test';
  const reserved = false;
  const handleToggleClick = () => true;

  const { getByTestId } = render(<Rocket
    id={id}
    rocket_name={rocketName}
    rocket_image={rocketImage}
    description={description}
    reserved={reserved}
    handleToggleClick={handleToggleClick}
  />);
  expect(getByTestId('rocket-btn')).toHaveTextContent('Reserve Rocket');
  const tree = renderer.create(<Rocket
    id={id}
    rocket_name={rocketName}
    rocket_image={rocketImage}
    description={description}
    reserved={reserved}
    handleToggleClick={handleToggleClick}
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
