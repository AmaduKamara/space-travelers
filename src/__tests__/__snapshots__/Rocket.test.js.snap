// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should do render reserve rocket button and match the snapshot 1`] = `
<div
  className="flex mb-6"
>
  <img
    alt="rocket"
    src="img"
    width="250"
  />
  <div
    className="mx-5"
  >
    <h2
      className="text-xl font-semibold pt-2"
    >
      Test Rocket
    </h2>
    <p
      className="my-4"
    >
       
      Test
    </p>
    <button
      className="py-2 px-5 rounded text-white bg-blue-500 shadow-md"
      data-testid="rocket-btn"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;
